### PR TITLE
Several upgrades and improvements to CI

### PR DESCRIPTION
**Pull request by tripu**
Looking at #217 I noticed that there was no info on the PR about Travis builds and the results of tests. I think that's because we're still on the (now deprecated) `travis-ci.org`, instead of the newer `travis-ci.com`.

I fixed that, plus a few other old shortcomings:

* `travis-ci.org` &rarr; `travis-ci.com`
* On Travis, build with latest Ubuntu (`Focal`)
* On Travis, build with both Node.js `16` &amp; `18`, and latest `npm`
* Use `npm ci` instead of `npm i`; add the build step to the CI cycle, too
* Specify these new Node.js &amp; npm requirements in README and enforce them in field `engines` of `package.json`
* Rebuild lock file with modern npm (`"lockfileVersion": 2`)
* Move default branch to `master`
* Upgrade all upgradeable _devDependencies_

After merging this, I'd move the default branch here from `gh-pages` to `master` (the only reason it was `gh-pages` was that at the time that was the only branch that GH Pages was able to serve, but that has changed now).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+  - lts/*
 branches:
   only:
-    - gh-pages
-
-cache:
-  directories:
-    - node_modules
-
+    - master
```
Comment (tristen) [inline .travis.yml:8]: 👍 can we make this `main` instead?

**Review by tristen — CHANGES_REQUESTED** [review]
👋 @tripu I might be missing something but should Travis be running against this branch? I don't see the status of tests against the PR.

**Comment by tripu** [comment]
> wave @tripu I might be missing something but should Travis be running against this branch? I don't see the status of tests against the PR.

I set up the Travis config so that only `master` is built. To save resources and avoid builds in all branches. But we can remove those two lines and have builds of everything, that's fine by me!

File: .travis.yml (hunk 1 above)
Comment (tripu) [reply .travis.yml:8]: I'd rather not. `master` is the historical standard. I see GH changed its default to `main` for new repos, but in Git it's still `master`.

**Review by tristen — APPROVED** [review]

**Comment by tripu** [comment]
Thank you, @tristen! :partying_face:

**Pull request merged**